libraries/rib-base/src/androidTest/java/com/badoo/ribs/android/dialogkeyboard/AlertDialogKeyboardTest.kt

```diff
@@ -0,0 +1,79 @@
+package com.badoo.ribs.android.dialogkeyboard
+
+import android.content.Context
+import android.view.inputmethod.InputMethodManager
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.matcher.ViewMatchers.withClassName
+import androidx.test.platform.app.InstrumentationRegistry
+import com.badoo.ribs.core.Rib
+import com.badoo.ribs.core.modality.BuildContext
+import com.badoo.ribs.core.modality.BuildParams
+import com.badoo.ribs.routing.source.backstack.BackStack
+import com.badoo.ribs.routing.source.backstack.operation.pushOverlay
+import com.badoo.ribs.test.RibsRule
+import com.badoo.ribs.test.util.ribs.root.TestRoot
+import com.badoo.ribs.test.util.ribs.root.TestRootRouter
+import com.badoo.ribs.test.util.ribs.root.builder.TestRootBuilder
+import com.badoo.ribs.test.util.runOnMainSync
+import com.badoo.ribs.test.util.waitFor
+import org.hamcrest.CoreMatchers.endsWith
+import org.junit.Rule
+import org.junit.Test
+import java.util.*
+
+class AlertDialogKeyboardTest {
+
+    @get:Rule
+    val ribsRule = RibsRule()
+
+    @Test
+    fun keyboardIsShownForRIBDialogWithEditText() {
+        setupRibs()
+        onView(withClassName(endsWith("EditText"))).perform(click())
```

Comment:
Why do you need a click, when keyboard should be displayed automatically?

Reply:
Keyboard will not be shown automatically if the view doesn't have focus.
I think that it's safer to make click here comparing to requesting focus in some other place.

The test is checking that the keyboard will be shown and not blocked from presenting, but not that keyboard will be shown automatically.
I will update test name to avoid confusions.